### ISSUE TITLE
Nested record generated when TABLE.COLUMN "overlaps" with a table called TABLE_COLUMN for MySQL

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#5334]
This has been reported as a regression in jOOQ 3.8.

----
See also:
https://groups.google.com/forum/#!topic/jooq-user/xnSC2_QHqD8

**Comment by Bob565656** [comment]
Traced it down to the following line:
`org.jooq.util.JavaGenerator [line 3100]`
`final String columnType = out.ref(getJavaType(column.getType()));`

Digging deeper, it seems the type is set incorrectly on
`org.jooq.util.JavaGenerator [line 4968]`
```
// [#3942] PostgreSQL treats UDTs and table types in similar ways
else if (db.getTable(schema, u) != null) {
    type = getStrategy().getFullJavaClassName(db.getTable(schema, u), udtMode);
}
```
I am using a MariaDB dialect, and a  postgress UDT was triggered

**Comment by Bob565656** [comment]
Problem was by adding a forced type the problem column
```
new Database()
.withForcedTypes(new ForcedType()
        .withName("TEXT")
        .withExpression("my_db\\.customer\\.notes"));
```

**Comment by lukaseder** [comment]
Thank you very much for reporting this and for all the details! This is indeed a 3.8 regression due to https://github.com/jOOQ/jOOQ/issues/3942. It affects only MySQL and MariaDB.

jOOQ internally generates synthetic user defined types for MySQL enums. These types have names of the form `tablename_columnname`. With the change in #3942, there is now an accidental match in these cases.

**Comment by lukaseder** [comment]
This is fixed on master and will be merged to 3.8.3 soon (https://github.com/jOOQ/jOOQ/issues/5339). If you need a more thorough workaround right now, please find the following commit, which you can apply as a patch to your version:
https://github.com/jOOQ/jOOQ/commit/a0f55fa4b89e13786f4b679e7a21c41b6f1ede56

Thanks again for reporting this!
